tests/ManiaTemplates.Tests/Lib/MtTransformerTest.cs

```diff
@@ -0,0 +1,114 @@
+            new() { Content = "scriptText2", Main = false, Once = true },
+            new() { Content = "scriptText3", Main = false, Once = false }
+        },
+        HasSlot = true,
+        ImportedComponents =
+            new()
+            {
+                { "component", new() { TemplateKey = "component.mt", Tag = "component" } },
+                { "Graph", new() { TemplateKey = "newGraph.mt", Tag = "Graph" } }
+            },
+        TemplateContent = @"<Frame if=""enabled"" foreach=""int i in numbers"" x=""{{ 20 * __index }}"">
+                                <Label if=""i &lt; numbers.Count"" foreach=""int j in numbers.GetRange(0, i)"" text=""{{ i }}, {{ j }} at index {{ __index }}, {{ __index2 }}"" />
+                            </Frame>
+                            <Frame>
+                                <Frame>
+                                    <test>
+                                        <Graph arg3=""{{ new test() }}""/>
+                                    </test>
+                                </Frame>
+                            </Frame>"
+    };
+
+    private readonly MtTransformer _transformer;
+
+    public MtTransformerTest()
+    {
+        _transformer = new MtTransformer(_maniaTemplateEngine, new MtLanguageT4());
+    }
+
+    [Fact]
+    public void Should_Build_Manialink()
+    {
+        var components = new Dictionary<string, MtComponent>
+        {
+            {
+                "newGraph.mt",
+                new MtComponent
+                {
+                    Namespaces = new() { "GraphNamespace" },
+                    Properties =
+                        new()
+                        {
+                            { "arg1", new() { Name = "arg1", Type = "string", Default = "" } },
+                            { "arg2", new() { Name = "arg2", Type = "int", Default = "0" } },
+                            { "arg3", new() { Name = "arg3", Type = "test" } }
+                        },
+                    Scripts = new List<MtComponentScript>()
+                    {
+                        new() { Content = "GraphScript", Main = false, Once = false },
+                        new() { Content = "GraphScript", Main = false, Once = true }
+                    },
+                    HasSlot = false,
+                    ImportedComponents = new MtComponentMap(),
+                    TemplateContent = ""
+                }
+            }
+        };
+        _maniaTemplateEngine.GetType().GetField("_components", BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.SetValue(_maniaTemplateEngine, components);
+
+        var sanitizedExpected = StripLineBreaks(File.ReadAllText("Lib/expected.tt"));
+        var result = _transformer.BuildManialink(_testComponent, "expected");
+        var sanitizedResult = StripLineBreaks(TransformCodeToOrderNumber(result));
+
+        Assert.Equal(sanitizedExpected, sanitizedResult);
+    }
+
+    private static string StripLineBreaks(string input)

```

Comment:
Is there a reason to ignore line breaks, other than different line breaks for windows and unix bases machines. We could uses xunit `Assert.equals`  named parameter `ignoreLineEndingDifferences` for that use case.

Reply:
Yeah probably the better solution, the comparison kept failing because of different OS line endings.